extension/lib/webcompat.js

```diff
@@ -0,0 +1,277 @@
+
+        if (unsupportedBrowsers.length || deprecated || experimental) {
+          issues.push({
+            type: _ISSUE_TYPE.CSS_PROPERTY_ALIASES,
+            aliases,
+            property,
+            deprecated,
+            experimental,
+            unsupportedBrowsers: unsupportedBrowsers.length ? unsupportedBrowsers : null,
+          });
+        }
+      }
+    }
+
+    return issues;
+  }
+
+  _asFloatVersion(version = false) {
+    if (version === true) {
+      return 0;
+    }
+    return version === false ? Number.MAX_VALUE : parseFloat(version);
+  }
+
+  _findAliases(aliases, property, compatData, context) {
+    if (compatData.__compat) {
+      for (let browser in compatData.__compat.support) {
+        const supportStates = compatData.__compat.support[browser] || [];
+        for (const { alternative_name, prefix }
+          of Array.isArray(supportStates) ? supportStates : [supportStates]) {
+          if (!prefix && !alternative_name) {
+            continue;
+          }
+
+          const alias = alternative_name || prefix + property;
+          aliases.push({ alias, aliasOf: property });
+        }
+      }
+
+      return;
+    }
+
+    for (let field in compatData) {
+      if (field.endsWith("_context")) {
+        this._findAliases(aliases, property, compatData[field], field);
+      }
+    }
+  }
+
+  _flattenAliases(node) {
+    const aliases = [];
+    for (let key in node) {
+      if (key.includes("_")) {
+        continue;
+      }
+      this._findAliases(aliases, key, node[key]);
+    }
+
+    for (const { alias, aliasOf } of aliases) {
+      if (!node[alias]) {
+        node[alias] = {};
+      }
+      node[alias]._aliasOf = aliasOf;
+      // Want to handle the entity property as same as alias property.
+      node[aliasOf]._aliasOf = aliasOf;
+    }
+  }
+
+  _getAlias(root, ...terms) {
+    const node = this._getNode(root, terms);
+    return node ? node._aliasOf : null;
+  }
+
+  _getChildNode(name, parent) {
+    name = name.toLowerCase();
+
+    let child = parent[name];
+
+    if (!child) {
+      for (let field in parent) {
+        if (name.startsWith(field)) {
+          child = parent[field];
+          break;
+        }
+      }
+
+      if (!child) {
+        return null;
+      }
+    }
+
+    if (child._aliasOf) {
+      child = parent[child._aliasOf]
+    }
+
+    return child;
+  }
+
+  _getCompatTable(root, terms) {
+    let node = this._getNode(root, terms);
+
+    if (!node) {
+      return null;
+    }
+
+    if (!node.__compat) {
+      for (let field in node) {
+        // TODO: We don't have a way to know the context for now.
+        //       Thus, we choose first context if need.
+        if (field.endsWith("_context")) {
+          node = node[field];
+          break;
+        }
+      }
+    }
+
+    return node.__compat;
+  }
+
+  _getNode(node, terms) {
+    for (const term of terms) {
+      node = this._getChildNode(term, node);
```

Comment:
I'm sorry for making you confused.

Actually, for now, as we handle only CSS property, it was enough by one term.
The reason why the parameter is multiple term is for next step such as CSS value.
For example, when we want to check the compatibility of CSS value `background-clip:  content-box;`, `_getNode` will receive the parameters like `_getNode(webCompatData.css.properties, ["background-clip", "content-box"])`.
The first parameter `webCompatData.css.properties` is JSON and is a tree which has all css property. So we can get a node for `background-clip` from the node. And as CSS value belongs to CSS property node, we can get a node for `content-box` from the CSS property node. ( https://github.com/mdn/browser-compat-data/blob/master/css/properties/background-clip.json#L112 )

Thus, although the function does more complex operation, but if I will say simply, the function returns `webCompatData.css.properties["background-clip"]["content-box"]`.

I'll add such the explanation as comment. Also, as you pointed, since the name of `node` is not clear the meaning, I will rename from `node` to something like `compatNode`

Reply:
I figured out after reading the whole code that `terms` is an array of keys to do depth traversal of nodes in the web compat data store. The pattern is used elsewhere in the file too.

Having code comments for each method to explain what `node` is (a node in the web compat data store) and what `terms` are (an array keys of nested objects within `node`) will better prepare future readers.